### PR TITLE
DISPATCH-1870 Do not test gRPC if HTTP/2 support was not built

### PR DESCRIPTION
**Pull request by jiridanek**
CC @fgiorgetti

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -65,6 +65,8 @@ message(STATUS "Found Proton: ${Proton_LIBRARIES} (found version \"${Proton_VERS
 
 # http2 support is optional
 find_package(libnghttp2 1.4.0)
+CMAKE_DEPENDENT_OPTION(USE_LIBNGHTTP2 "Use libnghttp2 for HTTP/2 support" ON
+                         "libnghttp2_FOUND" OFF)
```
Comment (jiridanek) [inline CMakeLists.txt:69]: @ganeshmurthy I think this is worthwhile change. It is the same pattern that is used for libwebsockets defined below this.

The name of the variable could be better, maybe? Something like `QD_USE_LIBNGHTTP2` or maybe `QD_ENABLE_HTTP2`? But then i'd have to rename the other one too, and would have to create alias for backwards compatibility....

Comment (ganeshmurthy) [reply CMakeLists.txt:69]: Can you please elaborate on why this change is needed ? Thanks.

Comment (jiridanek) [reply CMakeLists.txt:69]: It is the same pattern that is used for libwebsockets defined below this. It is not needed, but I think that Alan did nice work about CMake definitions for libwebsockets.

The reason for not using `libnghttp2_FOUND` everywhere in the CMakeLists.txt is that "the decision to use the library in compiling dispatch does not only depend on whether the library is present, but the user should also have the option to choose whether the library should be used".

Comment (jiridanek) [reply CMakeLists.txt:69]: OK, new PR, new Jira.

**Review by ganeshmurthy — APPROVED** [review]

**Pull request merged**